### PR TITLE
Tests for collections unification

### PR DESCRIPTION
**Pull request by dfilaretti**
Add some tests for collection unification. 

###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`
- [ ] Style conformance: `stylish-haskell`

---

### CODE REVIEW

**Comment by dfilaretti** [comment]
Jenkins: test this please

**Comment by dfilaretti** [comment]
@traiansf I changed the names of the folders and files of each individual test to give a bit more information about what each test is for.

@ttuegel added the email address now. For some reason I was assuming it was all set up already.

**Comment by dfilaretti** [comment]
Are we happy with this tests for the time being or would you like me to add more? 
Is there any case you are particularly interested in?

File: src/main/k/in-progress/tests/collections/list/list-unify-any-elem/Makefile [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+TOP != git rev-parse --show-toplevel
 include $(TOP)/include.mk
 
-KOMPILED := list-unify-1-kompiled
+KOMPILED := list-unify-any-elem-kompiled
 DEFINITION := $(KOMPILED)/definition.kore
 
-$(DEFINITION): list-unify-1.k $(KOMPILE_TARGETS)
-	$(KOMPILE) $(KOMPILE_OPTS) $< --syntax-module list-unify-1
+$(DEFINITION): list-unify-any-elem.k $(KOMPILE_TARGETS)
```
Comment (ttuegel) [inline src/main/k/in-progress/tests/collections/list/list-unify-any-elem/Makefile]: Please make this replacement in all the `Makefile`s:
```suggestion
$(DEFINITION): list-unify-any-elem.k $(K)
```

File: src/main/k/in-progress/tests/collections/list/list-unify-any-elem/Makefile [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
+	$(KOMPILE) $(KOMPILE_OPTS) $< --syntax-module list-unify-any-elem
 
-%.list-unify-1.kore: %.list-unify-1 $(DEFINITION) $(KRUN_TARGETS)
+%.list-unify-any-elem.kore: %.list-unify-any-elem $(DEFINITION) $(KRUN_TARGETS)
```
Comment (ttuegel) [inline src/main/k/in-progress/tests/collections/list/list-unify-any-elem/Makefile]: Please make this replacement in all the `Makefile`s:
```suggestion
%.list-unify-any-elem.kore: %.list-unify-any-elem $(DEFINITION) $(K) $(KORE_EXEC)
```

**Comment by traiansf** [comment]
The overall structure looks OK to me.

**Pull request merged**